core/common/test/format/DateTimeComponentsFormatTest.kt

```diff
@@ -335,42 +318,128 @@ class DateTimeComponentsFormatTest {
-        TimezoneTestData.tzPrefixes.forEach(::assertParseableAsTimeZone)
+    fun testSpecialNamedTimeZones() {
+        OffsetTimeZoneTestData.tzPrefixes.forEach(::assertParseableAsTimeZone)
     }
 
     @Test
     fun testPrefixWithCorrectParsableOffset() {
-        val timezoneIds =
-            generateTimezoneIds(TimezoneTestData.tzPrefixes + "", TimezoneTestData.correctParsableOffsets)
+        val timezoneIds = generateTimeZoneIds(
+            OffsetTimeZoneTestData.tzPrefixes + "",
+            OffsetTimeZoneTestData.correctParsableOffsets
+        )
         timezoneIds.forEach(::assertParseableAsTimeZone)
     }
 
     @Test
     fun testPrefixWithIncorrectParsableOffset() {
-        val timezoneIds =
-            generateTimezoneIds(TimezoneTestData.tzPrefixes + "", TimezoneTestData.incorrectParsableOffsets)
+        val timezoneIds = generateTimeZoneIds(
+            OffsetTimeZoneTestData.tzPrefixes + "",
+            OffsetTimeZoneTestData.incorrectParsableOffsets
+        )
         timezoneIds.forEach(::assertIncorrectlyParseableAsTimeZone)
     }
 
+    private object NamedTimeZoneTestData {
+        val validSinglePart = listOf(
+            // Simple names
+            "EST", "PST", "CET", "CEST",
+            // Names with underscores
+            "Eastern_Standard", "Pacific_Standard_Time",
+            // Names with dots
+            "US.Eastern", "Australia.Sydney",
+            // Names with digits
+            "GMT0", "UTC8", "Zone1",
+            // Names with hyphens
+            "GMT-Special", "UTC-Extended",
+            // Names with plus signs
+            "GMT+Special", "Zone+Extended",
+            // Mixed characters
+            "Zone_1.2-Special+Extended",
+            // Starting with a dot or underscore
+            ".SpecialZone", "_InternalZone"
+        )
+
+        val validMultiPart = listOf(
+            // Common real-world timezones
+            "America/New_York", "Europe/London", "Asia/Tokyo", "Australia/Sydney", "Africa/Cairo",
+            // Three parts
+            "America/Indiana/Indianapolis", "America/Argentina/Buenos_Aires", "America/Kentucky/Louisville", "_/./a",
+            // Many parts
+            "A/B/C/D/E/F/G/H/I/J/KLM/N/O/P/Q/R/S/T/U/V/W/Z/Y/Z",
+            // Multiple parts with various characters
+            "Etc/GMT+1", "Etc/GMT-10", "US/East-Indiana", "Brazil/DeNoronha",
+            // With dots
+            "System.Default/Zone.1",
+            // Complex names
+            "Region_1/Sub-Region.2/Zone+3",
+            // Maximum complexity
+            "Region_1.2-3+4/Sub-Region_5.6-7+8/Zone_9.0-1+2"
+        )
+
+        val invalid = listOf(
+            // Empty string
+            "",
+            // Starting with invalid characters
+            "1America", "-Zone", "+Zone", "/Zone",
+            // Starting with slash
+            "/America/New_York",
+            // Ending with slash
+            "America/New_York/", "America/",
```

Comment:
It's interesting, though, that even though it doesn't get parsed, the trailing slash does not make the string incorrect. I think this makes sense, and it's worth testing this behavior (to keep track of it).

Reply:
Yes, this.